parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -1127,7 +1127,7 @@ public void appendColumnChunk(ColumnDescriptor descriptor, SeekableInputStream f
     long length = chunk.getTotalSize();
     long newChunkStart = out.getPos();
 
-    if (newChunkStart != start) {
+    if (offsetIndex != null && newChunkStart != start) {
```

Comment:
Good catch! Could you please add a test case to make sure this is well covered now?

Reply:
Found a file without ColumnIndexes that was added for another test case - if you have any advice how it could be generated instead I'd be happy to hear but it looks prohibitive to do so with the current implementation.

Reply:
Thank you for adding test!

BTW, PARQUET-2081 is a fixed issue for a new feature. Would you mind opening a new JIRA for this bug fix?